sdk/search/test-resources.json

```diff
@@ -48,6 +48,14 @@
     "ENDPOINT": {
       "type": "string",
       "value": "[concat('https://', parameters('baseName'), '.search.windows.net')]"
+    },
+    "CHINAENDPOINT": {
+      "type": "string",
+      "value": "[concat('https://', parameters('baseName'), '.search.azure.cn')]"
+    },
+    "USENDPOINT": {
+      "type": "string",
+      "value": "[concat('https://', parameters('baseName'), '.search.azure.us')]"
```

Comment:
@benbp Is this what you had in mind when logging #17577?

Here we are retaining the switch statement on the authority host to pick the different endpoints from the env values set by the template. I was under the impression that we should be doing away with such switch statements completely and rely on the template to pass the right endpoints. Am I missing something?

Reply:
@sarangan12 Please sync with @benbp on the above

Reply:
Sure. I will sync with @benbp in the morning

Reply:
@ramya-rao-a @sarangan12 These values get passed in via a [cloud configuration JSON](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/TestResources/clouds/AzureChinaCloud.json#L11) and handled by the deployment script, where it will get passed as an ARM template parameter during deployment. The recommended way to do this is to take the ARM template parameter and set whatever environment variable you like from it.

For example, see the .net template [parameter declaration](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/test-resources.json#L44-L50) and [env var output declaration](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/search/test-resources.json#L179-L182).

So if you wanted to construct the full URL, it would also be simpler to do it in the template, similar to what's been done in this PR but using a parameter:

```
        "SEARCH_ENDPOINT": {
            "type": "string",
            "value": "[concat('https://', parameters('baseName'), '.', parameters('searchEndpointSuffix'))]"
        },```